server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -431,10 +436,15 @@ public DynamicReference getDynamic(ColumnIdent ident, boolean forWrite) {
 
         switch (parentPolicy) {
             case DYNAMIC:
-                if (!forWrite) return null;
+                if (!forWrite) {
+                    if (errorOnUnknownObjectKey == false) {
+                        throw new RuntimeException("throw in order to unwind the nest calls to sourceRelation.getField");
+                    }
```

Comment:
Not sure if I like this. It couples `DocTableInfo.getDynamic` further to how other components use it.
If we really need to use exceptions for flow control I think we should at least add a dedicated new exception and document this behavior - to make it part of the contract.

Reply:
I understand your concern that an exception for flow control is a bad practice. The case that requires this exceptional control flow is `select obj_strict['uknown_key'] from (select obj_strict from tbl) as ali` (although I need to confirm the expected behaviour). So, if it returns instead of throwing, `aliasedRelation.getField` will wrap the returned value as `scopedSymbol` which will lead to storage parsing.